### PR TITLE
DRAFT Compiler FE: Enable ScatterNd operator

### PR DESCRIPTION
**Pull request by kvochko**
This will implement ScatterND operation in compiler frontend

For #1711 

ONE-DCO-1.0-Signed-off-by: Andrey Kvochko <a.kvochko@samsung.com>

### CODE REVIEW

File: compiler/luci/import/src/Nodes/CircleScatterNd.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+bool CircleScatterNdGraphBuilder::validate(const ValidateArgs &args) const
+{
+  if (args.op.inputs.size() != 3)
+    return false;
+
+  return true;
+}
+
+CircleNode *CircleScatterNdGraphBuilder::build_node(const circle::OperatorT &op,
```
Comment (d-krylov) [inline compiler/luci/import/src/Nodes/CircleScatterNd.cpp:34]:
```suggestion
CircleNode *CircleScatterNdGraphBuilder::build_node(const circle::OperatorT &,
```

File: compiler/tflite2circle/src/BuildBuiltinOptions.h [hunk 2]
```diff
@@ -91,3 +91,4 @@
 #include "BuildBuiltinOptions/ZerosLikeOptions.h"
 
 #endif // __BUILD_BUITIN_OPTIONS_H__
+#include "BuildBuiltinOptions/ScatterNdOptions.h"
```
Comment (d-krylov) [inline compiler/tflite2circle/src/BuildBuiltinOptions.h:94]: I think, this line should be before
```
#include "BuildBuiltinOptions/SelectOptions.h"
```

File: compiler/luci/service/src/CircleShapeInferenceRule.cpp [hunk 3]
```diff
@@ -1383,6 +1383,31 @@ class ShapeInferenceAlgorithm final : public luci::CircleNodeVisitor<loco::NodeS
...
+
+    auto shape_node = loco::must_cast<luci::CircleConst *>(node->shape());
+
+    const loco::DataType S32 = loco::DataType::S32;
+    const loco::DataType S64 = loco::DataType::S64;
+
+    std::vector<int64_t> vect_shape;
+
+    if (shape_node->dtype() == S32)
+      vect_shape = vector_from_constant<S32>(shape_node);
```
Comment (seanshpark) [inline compiler/luci/service/src/CircleShapeInferenceRule.cpp:1398]: Thank you for using recently introduce method :)

File: res/TensorFlowPythonExamples/examples/scatter_nd/__init__.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# This example is a modified example from TF API documentation
```
Comment (seanshpark) [inline res/TensorFlowPythonExamples/examples/scatter_nd/__init__.py:1]: 👍

**Comment by seanshpark** [comment]
Looks good! Let's proceed with posting PRs :)

**Comment by kvochko** [comment]
All done

**Pull request closed without merging**